### PR TITLE
chore: upgrade to vitest

### PR DESCRIPTION
**Pull request by F-loat**
Upgrade to vitest and improve test coverage 

The two methods `bodyToString` and `formBodyToString` are not used, why should they be retained👀

image

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -26,26 +26,26 @@ type StringMap = { [key: string]: string };
...
-  location:           boolean,
-  locationTrusted:    boolean,
-  noKeepalive:        boolean,
-  output:             string,
-  showError:          boolean,
-  silent:             boolean,
-  ssl:                boolean,
-  sslv2:              boolean,
-  sslv3:              boolean,
-  verbose:            boolean,
```
Comment (albertodeago) [inline src/main.ts:48]: why would you change this types? 👀

Comment (albertodeago) [reply src/main.ts:48]: Thanks for taking time to open this, I'll try to have a deeper look and try it out as soon as I've some available time 🙏

Comment (F-loat) [reply src/main.ts:48]: `undefined` should be allowed for no such config, and `false` will cause an runtime error.
https://github.com/albertodeago/curl-generator/blob/bf1925e01a686ea03da99279f685e7463554c6e9/src/main.ts#L125-L126

Comment (F-loat) [reply src/main.ts]: These types of changes have been reverted and may be you can considered again later.

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+# top-most EditorConfig file
+root = true
+
+[*]
+indent_style = space
+indent_size = 2
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = false
+insert_final_newline = false
```
Comment (albertodeago) [inline .editorconfig:12]: I prefer not having the editorconfig
will probably install biome to take care of lint/formatting

File: test/main.test.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import { expect, test } from "vitest";
+import { CurlGenerator } from "../src/main";
+import * as params from "../example/parameters";
+
+Object.keys(params).forEach((key) => {
+  test(key, () => {
+    const result = CurlGenerator(params[key], params[key].options);
+    expect(result).toMatchSnapshot();
```
Comment (albertodeago) [inline test/main.test.ts:8]: nice, I like the approach of having the results as snapshots, make things cleaner

**Comment by albertodeago** [comment]
ah! @F-loat I see what you meant with the types.... Sorry I was actually wrong, the approach you were suggesting is indeed better... 
Maybe can put them all as optional boolean (e.g. `compressed?: boolean`), so one can actually explicitly set a param to false (even if would be the same as omitting it).

What do you think?

File: test/main.test.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    })
+  ).toThrowError(/^Invalid form body value type/);
+});
+
+test("error options", () => {
+  expect(() =>
+    CurlGenerator(
+      { url: "https://jsonplaceholder.typicode.com/posts/1" },
+      {
+        silent: false as unknown as true,
```
Comment (albertodeago) [inline test/main.test.ts:50]: if we fix the typings (as you were suggesting, my bad 🙏) we can get rid also of this cast

File: test/main.test.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+test("error body type", () => {
+  expect(() =>
+    CurlGenerator({
+      url: "https://jsonplaceholder.typicode.com/posts",
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: 123 as unknown as string,
```
Comment (albertodeago) [inline test/main.test.ts:20]: I believe this should be `"123"`, without the cast, right?

**Comment by albertodeago** [comment]
Minor things, then it's ready to merge! 🚀

**Comment by albertodeago** [comment]
Indeed `bodyToString` and `formBodyToString` looks not utilised (I believe they are a leftover of another PR)... We can get rid of them 🧹

**Comment by F-loat** [comment]
@albertodeago  It's all done, you can look at it again

**Comment by albertodeago** [comment]
Thank you @F-loat 
Merging and releasing a new patch version with this

**Review by albertodeago — APPROVED** [review]

**Pull request merged**